src/rebuild/srv.c

```diff
@@ -1378,6 +1383,103 @@ rebuild_print_list_update(const char *const str, const uuid_t uuid,
 	D_PRINT("\n");
 }
 
+/* Here we want to schedule this new rebuild task by placing it on the
+ * queue. However, the ordering is important - tasks will be consumed
+ * by iterating the queue and running the first available one for each
+ * pool. So what we do here is iterate through the queue, and insert
+ * this task before the first other found task for the same pool that
+ * should run with lower priority. By doing this, the list will always
+ * be sorted
+ */
```

Comment:
Can we do these rebuild, drain, reint, extend operations despite of f_seq / pool version order?

Reply:
This was an excellent catch, and indeed we can not do this - and it uncovered the actual bug here (which is that rebuild was allowed to merge tasks together in such a way as to make the versions incorrect. Thanks a bunch for catching this!